test/acl_auto_configure_test.cpp

```diff
@@ -261,6 +269,27 @@ TEST(auto_configure, simple) {
               (int)m_device_def.autodiscovery_def.accel[0].max_work_group_size);
   CHECK_EQUAL(1, (int)m_device_def.autodiscovery_def.accel[0].is_sycl_compile);
 
+  // Checks for device global entry.
+  CHECK_EQUAL(2, (int)m_device_def.autodiscovery_def.num_device_global);
+  CHECK(m_device_def.autodiscovery_def.device_global_mem_defs.find(
+            "kernel15_dev_global") !=
+        m_device_def.autodiscovery_def.device_global_mem_defs.end());
+  CHECK(m_device_def.autodiscovery_def.device_global_mem_defs.find(
+            "kernel15_dev_global2") !=
+        m_device_def.autodiscovery_def.device_global_mem_defs.end());
+  CHECK_EQUAL(4096, m_device_def.autodiscovery_def
+                        .device_global_mem_defs["kernel15_dev_global"]
```

Comment:
Instead of doing a write access using [`[]`](https://en.cppreference.com/w/cpp/container/unordered_map/operator_at), store the return values of [`find`](https://en.cppreference.com/w/cpp/container/unordered_map/find), then use them here to access the members. (This would not compile if `m_device_def` were `const`, hence constify all the things :slightly_smiling_face:).